doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,456 @@
+  * Removes planning constraints
+  **/
+  void removeConstraints()
+  {
+    // BEGIN_SUB_TUTORIAL remove_constraint
+    // After moving it is often useful to revert back to the default settings.
+    move_group_.clearPathConstraints();
+    // The demo should show two paths to compare constrained plan with an unconstrained plan.
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+   * This sets the planning time to a number of seconds.
+   **/
+  void setPlanningTime(double time = 5.0)
+  {
+    // BEGIN_SUB_TUTORIAL planning_time
+    // Planning with constraints can be slow because every sample must call an inverse kinematics
+    // solver. Lets increase the planning time from the default 5 seconds to be sure the planner
+    // has enough time to succeed.
+    move_group_.setPlanningTime(time);
+    // END_SUB_TUTORIAL
+  }
+  /**
+  * Populates the plan with a cartesian path following the given points in order
+  **/
+  double planCartesianPath(const std::vector<geometry_msgs::Pose>& target_poses,
+                           moveit::planning_interface::MoveGroupInterface::Plan& plan)
+  {
+    // BEGIN_SUB_TUTORIAL cartesian_plan
+    //
+    // Cartesian motions are frequently needed to be slower for actions such as approach and
+    // retreat grasp motions. Here we demonstrate how to reduce the speed of the robot arm via
+    // a scaling factor of the maxiumum speed of each joint. Note this is the joint velocities
+    // not the end effector velocities.
+    move_group_.setMaxVelocityScalingFactor(0.1);
+
+    // We want the Cartesian path to be interpolated at a resolution of 1 cm
+    // which is why we will specify 0.01 as the max step in Cartesian
+    // translation.  We will specify the jump threshold as 0.0, effectively disabling it.
+    // Warning - disabling the jump threshold while operating real hardware can cause
+    // large unpredictable motions of redundant joints and could be a safety issue.
+    moveit_msgs::RobotTrajectory trajectory;
+    const double jump_threshold = 0.0;
+    const double eef_step = 0.01;
+    double fraction = move_group_.computeCartesianPath(target_poses, eef_step, jump_threshold, trajectory);
+
+    // While we didn't use the planner to create this trajectory it still can be added to a plan
+    plan.trajectory_ = trajectory;
+    // Moving the robot with this plan is the same as before.
+    //
+    // When finished remember to reset the scalingfactor to return to normal speeds.
+    move_group_.setMaxVelocityScalingFactor(1);
+    // END_SUB_TUTORIAL
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan (Cartesian path) (%.2f%% acheived)", fraction * 100.0);
+    return fraction;
+  }
+
+  /**
+  * Adds the given object to the world
+  * shape_msgs::SolidPrimitive primitive - The shape
+  * geometry_msgs::Pose box_pose - The location
+  * std::string name - The object ID
+  **/
+  void createObject(const shape_msgs::SolidPrimitive& primitive, const geometry_msgs::Pose& box_pose, std::string name)
```

Comment:
function needs more specific name... it only creates boxes, right?

Reply:
The shape_msgs::SolidPrimitive can be a cone, sphere, square or cylinder. It is only used to create boxes but it could be used for other shaped object creation.